### PR TITLE
Put expected/actual types in error message

### PR DESCRIPTION
**Pull request by jez**
### Motivation

Better error messages

### Test plan

See included automated tests.

**Before**
Screen Shot 2020-12-22 at 11 56 59 PM

**After**
Screen Shot 2020-12-22 at 11 58 07 PM

### CODE REVIEW

**Review by elliottt — APPROVED** [review]
This is a much nicer error message!

File: infer/environment.cc [hunk 1]
```diff
@@ -1164,7 +1164,8 @@ core::TypePtr Environment::processBinding(core::Context ctx, const cfg::CFG &inW
                                                            core::UntypedMode::AlwaysCompatible)) {
                     if (auto e = ctx.beginError(bind.loc, core::errors::Infer::ReturnTypeMismatch)) {
                         auto ownerData = ctx.owner.data(ctx);
-                        e.setHeader("Returning value that does not conform to method result type");
+                        e.setHeader("Expected `{}` but found `{}` for method result type", methodReturnType.show(ctx),
```
Comment (Morriar) [inline infer/environment.cc:1167]:
```suggestion
                        e.setHeader("Expected return type to be `{}` but found `{}`", methodReturnType.show(ctx),
```

**Review by Morriar — COMMENTED** [review]
Nice!

File: infer/environment.cc (hunk 1 above)
Comment (jez) [reply infer/environment.cc:1167]: I was planning to factor this all out into a helper in a future PR, where the form is always “Expected X but found Y for Z” (this is already the form that argument type errors and others fall into).

@Morriar how strongly do you feel here?

Comment (Morriar) [reply infer/environment.cc:1167]: Not strongly at all, but when reading the error, I'd prefer to have the "*return* type" information first so I know what I need to think of before reading the expected type.

Comment (jez) [reply infer/environment.cc:1167]: Sounds good. I’ll leave it as is for now, and in the PR to make all the errors more consistent I’ll revisit it.

**Comment by przemyslaw2000** [comment]
#3468

**Pull request merged**